### PR TITLE
7180 potential race between zfs_resume_fs and zfs_ioc_rename

### PR DESCRIPTION
**Pull request by avg-I**
zfs_suspend_fs and zfs_resume_fs are called in the open context.
Because of that it's possible that the filesystem (the datatset)
gets renamed while it is suspended.  In that case we won't be able
to find the dataset by its old name.  On the other hand, we keep
a long hold on the dataset, so we can use it to refresh the objset
pointer.

### CODE REVIEW

**Comment by prakashsurya** [comment]
@zettabot go

File: usr/src/uts/common/fs/zfs/zfs_ioctl.c [hunk 1]
```diff
@@ -3641,16 +3641,18 @@ zfs_ioc_destroy(zfs_cmd_t *zc)
 static int
 zfs_ioc_rollback(const char *fsname, nvlist_t *args, nvlist_t *outnvl)
 {
+	dsl_dataset_t *ds;
```
Comment (ahrens) [inline usr/src/uts/common/fs/zfs/zfs_ioctl.c]: I'd prefer to declare this inside the `if` block to limit its scope.

Comment (avg-I) [inline usr/src/uts/common/fs/zfs/zfs_ioctl.c]: done

**Comment by prakashsurya** [comment]
@zettabot go

File: usr/src/uts/common/fs/zfs/zfs_vfsops.c [hunk 2]
```diff
@@ -2033,10 +2033,9 @@ zfs_resume_fs(zfsvfs_t *zfsvfs, const char *osname)
 	 * objset_t, as the one we had before may have been evicted.
```
Comment (pzakha) [inline usr/src/uts/common/fs/zfs/zfs_vfsops.c]: Please update the comment block

Comment (avg-I) [inline usr/src/uts/common/fs/zfs/zfs_vfsops.c]: Oh, my apologies.  I meant to make this fix, but then it slipped my mind.
I see that you have sent an RTI, is it pointless for me to update this PR now?
Or should I still do it?
Thank you!

**Pull request closed without merging**